### PR TITLE
Support GC zeal

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11154)

### CODE REVIEW

File: components/script/script_runtime.rs [hunk 1]
```diff
@@ -377,3 +379,23 @@ unsafe extern fn trace_rust_roots(tr: *mut JSTracer, _data: *mut os::raw::c_void
     trace_roots(tr);
     debug!("done custom root handler");
 }
+
+#[allow(unsafe_code)]
+#[cfg(feature = "debugmozjs")]
+unsafe fn set_gc_zeal_options(ctx: *mut JSContext) {
```
Comment (Ms2ger) [inline components/script/script_runtime.rs]: Nit: `cx`, not `ctx`.

File: components/script/script_runtime.rs [hunk 2]
```diff
@@ -377,3 +379,23 @@ unsafe extern fn trace_rust_roots(tr: *mut JSTracer, _data: *mut os::raw::c_void
...
+    use js::jsapi::{JS_DEFAULT_ZEAL_FREQ, JS_SetGCZeal};
+
+    let level = match get_pref("js.mem.gc.zeal.level").as_i64() {
+        Some(level @ 0...14) => level as u8,
+        _ => return,
+    };
+    let frequency = match get_pref("js.mem.gc.zeal.frequency").as_i64() {
+        Some(frequency) if frequency >= 0 => frequency as u32,
```
Comment (Ms2ger) [inline components/script/script_runtime.rs]: Can we use `0...` here?

**Comment by nox** [comment]
We can't use `0...` (nor `0..`), but I fixed the `cx` nit.

@bors-servo r=Ms2ger

**Pull request merged**